presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -153,30 +141,46 @@ public static Builder builder()
 
     public static Builder buildFrom(PlanNodeStatsEstimate other)
     {
-        return builder().setOutputRowCount(other.getOutputRowCount())
-                .setSymbolStatistics(other.symbolStatistics);
+        return new Builder(other.getOutputRowCount(), other.symbolStatistics);
     }
 
     public static final class Builder
     {
-        private double outputRowCount = NaN;
-        private Map<Symbol, SymbolStatsEstimate> symbolStatistics = new HashMap<>();
+        private double outputRowCount;
+        private PMap<Symbol, SymbolStatsEstimate> symbolStatistics;
+
+        public Builder()
+        {
+            this(NaN, HashTreePMap.empty());
+        }
+
+        private Builder(double outputRowCount, PMap<Symbol, SymbolStatsEstimate> symbolStatistics)
+        {
+            this.outputRowCount = outputRowCount;
+            this.symbolStatistics = symbolStatistics;
+        }
 
         public Builder setOutputRowCount(double outputRowCount)
         {
             this.outputRowCount = outputRowCount;
             return this;
         }
 
-        public Builder setSymbolStatistics(Map<Symbol, SymbolStatsEstimate> symbolStatistics)
+        public Builder addSymbolStatistics(Symbol symbol, SymbolStatsEstimate statistics)
         {
-            this.symbolStatistics = new HashMap<>(symbolStatistics);
+            symbolStatistics = symbolStatistics.plus(symbol, statistics);
             return this;
         }
 
-        public Builder addSymbolStatistics(Symbol symbol, SymbolStatsEstimate statistics)
+        public Builder addSymbolStatistics(Map<Symbol, SymbolStatsEstimate> symbolStatistics)
+        {
+            this.symbolStatistics = this.symbolStatistics.plusAll(symbolStatistics);
+            return this;
+        }
+
+        public Builder removeSymbolStatistics(Symbol symbol)
```

Comment:
Maybe, but you need to construct initial `PlanNodeStatsEstimate` for which the builder seems to be useful